src/librados/librados_c.cc

```diff
@@ -2898,6 +2898,81 @@ extern "C" int _rados_notify2(rados_ioctx_t io, const char *o,
 }
 LIBRADOS_C_API_BASE_DEFAULT(rados_notify2);
 
+extern "C" int _rados_decode_notify_msg(char *reply_buffer, size_t reply_buffer_len,
+                                        struct notify_ack_t **acks, size_t *nr_acks,
+                                        struct notify_timeout_t **timeouts, size_t *nr_timeouts) {
+  if (!reply_buffer || !reply_buffer_len) {
+    return -EINVAL;
+  }
+
+  bufferlist bl;
+  try{
+    bufferptr p = buffer::create(reply_buffer_len);
+    memcpy(p.c_str(), reply_buffer, reply_buffer_len);
+    bl.push_back(p);
+  } catch (const ceph::buffer::error &e) {
+    return -EINVAL;
+  }
```

Comment:
```suggestion
   bl.append(reply_buffer, c);
```

or if you really want to avoid `memcpy()`, please use
```c++
bl.append(ceph::buffer::ptr{ceph::buffer::create_static(create_static, reply_buffer)});